elftools/ehabi/ehabiinfo.py

```diff
@@ -0,0 +1,212 @@
+                # highest half must be 0b1000 for compact model
+                if word0 & 0x70000000 != 0:
+                    print ('Corrupt ARM compact model table entry: %x' % n)
+                    return CorruptEHABIEntry()
+                per_index = (word0 >> 24) & 0x7f
+                if per_index == 0:
+                    # arm compact model 0
+                    opcode = [(word0 & 0xFF0000) >> 16, (word0 & 0xFF00) >> 8, word0 & 0xFF]
+                    return EHABIEntry(function_offset, per_index, opcode)
+                elif per_index == 1 or per_index == 2:
+                    # arm compact model 1/2
+                    more_word = (word0 >> 16) & 0xff
+                    opcode = [(word0 >> 8) & 0xff, (word0 >> 0) & 0xff]
+                    self._arm_idx_section.stream.seek(eh_table_offset + 4)
+                    for i in range(more_word):
+                        r = struct_parse(self._struct.EH_table_struct, self._arm_idx_section.stream)['word0']
+                        opcode.append((r >> 24) & 0xFF)
+                        opcode.append((r >> 16) & 0xFF)
+                        opcode.append((r >> 8) & 0xFF)
+                        opcode.append((r >> 0) & 0xFF)
+                    return EHABIEntry(function_offset, per_index, opcode, eh_table_offset=eh_table_offset)
+                else:
+                    print ('Unknown ARM compact model %d at table entry: %x' % (per_index, n))
+                    return CorruptEHABIEntry()
+        else:
+            # highest bit is one, compact model must be 0
+            if word1 & 0x7f000000 != 0:
+                print('Corrupt ARM compact model table entry: %x' % n)
+                return CorruptEHABIEntry()
+            opcode = [(word1 & 0xFF0000) >> 16, (word1 & 0xFF00) >> 8, word1 & 0xFF]
+            return EHABIEntry(function_offset, 0, opcode)
+
+
+class EHABIEntry(object):
+    """ Exception handler abi entry.
+
+        Accessible attributes:
+
+            function_offset:
+                Integer.
+                Return None when corrupt.
+
+            personality:
+                Integer.
+                Return None when corrupt or unwindable.
+                0/1/2 for ARM personality compact format.
+                Others for generic personality.
+
+            bytecode_array:
+                Integer array.
+                Return None when corrupt or unwindable.
+
+            eh_table_offset:
+                Integer.
+                Return None when corrupt or unwindable or ARM inline compact.
+
+            unwindable:
+                bool. Whether this function is unwindable.
+
+            corrupt:
+                bool. Whether this entry is corrupt.
+
+    """
+
+    def __init__(self,
+                 function_offset,
+                 personality,
+                 bytecode_array,
+                 eh_table_offset=None,
+                 unwindable=True,
+                 corrupt=False
+                 ):
```

Comment:
I find this line wrapping a little odd, compared to methods like `CorruptEHABIEntry` that seem to wrap somewhere between 80-100 chars.  Maybe I'm just on a smaller screen now,  but `__init__` vs other places has what seems like inconsistent line wrapping.

Reply:
I designed in this format just for readable, rather than for code style. Do you think it really matters?

Reply:
maybe @eliben has thoughts?

Reply:
pyelftools tries to roughly follow https://www.python.org/dev/peps/pep-0008/ and be consistent. I think this wrapping is OK overall, but do place the closing `):` higher up (`corrupt=False):`), since the declaration is already visually distinct from the code below it (different indentation level, following the "wrapping" advice in PEP 8)